File: tests/java-rosettacode/Palindrome.java
public class Palindrome {
    public static boolean isPalindrome(String s) {
        int left = 0, right = s.length() - 1;
        while (left < right) {
            if (s.charAt(left) != s.charAt(right)) return false;
            left++;
            right--;
        }
        return true;
    }

    public static void main(String[] args) {
        String[] tests = {"racecar", "hello", "madam", "world", "level"};
        for (String s : tests) {
            System.out.printf("%s: %b%n", s, isPalindrome(s));
        }
    }
}

File: tests/rosettacode/palindrome.go
package main

import "fmt"

func isPalindrome(s string) bool {
	n := len(s)
	for i := 0; i < n/2; i++ {
		if s[i] != s[n-1-i] {
			return false
		}
	}
	return true
}

func main() {
	words := []string{"racecar", "hello", "level", "madam"}
	for _, w := range words {
		fmt.Printf("%s: %t\n", w, isPalindrome(w))
	}
}
